sql: gate certain box2d comparison ops by a cluster setting

otan opened the pull request:
Release note (sql change): box2d comparison operators are now gated by
the cluster setting
`sql.spatial.experimental_box2d_comparison_operators.enabled`.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52990)
<!-- Reviewable:end -->

otan committed:
sql: gate certain box2d comparison ops by a cluster setting

Release note (sql change): box2d comparison operators are now gated by
the cluster setting
`sql.spatial.experimental_box2d_comparison_operators.enabled`.

rytaft reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

Do you think it's worth also adding a session setting?

> Reviewed 1 of 3 files at r1, 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52990#-:-MF2lx6s1XI1oVixnAfX:b-pv8864)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
i tried that initially -- the session setting doesn't help if the parameter operates on a remote processor AFAICT.

rytaft commented:
Did you mean to increment the server version here? Or was that meant for another PR?

> i tried that initially -- the session setting doesn't help if the parameter operates on a remote processor AFAICT.

That's surprising... but I don't know enough about how these settings are passed around so I'll take your word for it. The cluster setting is more important anyway.





<!-- Sent from Reviewable.io -->

otan commented:
oops, yeah. thanks for the catch.

rytaft reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 2 files at r3, 4 of 4 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52990#-:-MF34Ew15jFi1TkCnT_V:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
bors r=rytaft
